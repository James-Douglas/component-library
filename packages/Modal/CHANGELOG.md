# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.8.3...@comparethemarketau/manor-modal@0.8.4) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.8.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.8.2...@comparethemarketau/manor-modal@0.8.3) (2020-09-28)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.8.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.8.1...@comparethemarketau/manor-modal@0.8.2) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.8.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.8.1...@comparethemarketau/manor-modal@0.8.2) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.8.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.8.0...@comparethemarketau/manor-modal@0.8.1) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.8.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.7.0...@comparethemarketau/manor-modal@0.8.0) (2020-09-24)


### Features

* **deps:** update all deps again ([#404](https://github.com/comparethemarketau/manor-react/issues/404)) ([7ac9088](https://github.com/comparethemarketau/manor-react/commit/7ac9088c9f042a656b2e8cb2bab615ed44bad88c))





# [0.7.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.6.2...@comparethemarketau/manor-modal@0.7.0) (2020-09-24)


### Features

* **rollup:** update rollup to not bundle manor-provider ([#403](https://github.com/comparethemarketau/manor-react/issues/403)) ([0e012d6](https://github.com/comparethemarketau/manor-react/commit/0e012d6fbadcf0ec99857c22e148cacd6265b60a))





## [0.6.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.6.1...@comparethemarketau/manor-modal@0.6.2) (2020-09-24)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.6.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.6.0...@comparethemarketau/manor-modal@0.6.1) (2020-09-23)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.6.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.11...@comparethemarketau/manor-modal@0.6.0) (2020-09-22)


### Features

* **provider:** provider context now contains breakpoint & isDesktop ([#398](https://github.com/comparethemarketau/manor-react/issues/398)) ([611b52f](https://github.com/comparethemarketau/manor-react/commit/611b52f8883ca431523b0a8896d51e66ecb76376))





## [0.5.11](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.10...@comparethemarketau/manor-modal@0.5.11) (2020-09-21)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.9...@comparethemarketau/manor-modal@0.5.10) (2020-09-20)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.8...@comparethemarketau/manor-modal@0.5.9) (2020-09-18)


### Bug Fixes

* **dependencies:** fix more incorrect dependencies ([#392](https://github.com/comparethemarketau/manor-react/issues/392)) ([6fa55a1](https://github.com/comparethemarketau/manor-react/commit/6fa55a11ba89125ccfe61385d9776e4185bff6f3))





## [0.5.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.7...@comparethemarketau/manor-modal@0.5.8) (2020-09-18)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.6...@comparethemarketau/manor-modal@0.5.7) (2020-09-17)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.5...@comparethemarketau/manor-modal@0.5.6) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.4...@comparethemarketau/manor-modal@0.5.5) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.3...@comparethemarketau/manor-modal@0.5.4) (2020-09-09)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.2...@comparethemarketau/manor-modal@0.5.3) (2020-09-08)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.1...@comparethemarketau/manor-modal@0.5.2) (2020-09-04)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.5.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.5.0...@comparethemarketau/manor-modal@0.5.1) (2020-09-03)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.5.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.4.2...@comparethemarketau/manor-modal@0.5.0) (2020-09-01)


### Features

* **notification:** added custom notification icon ([#368](https://github.com/comparethemarketau/manor-react/issues/368)) ([275c407](https://github.com/comparethemarketau/manor-react/commit/275c4071b225634fe0b13be576a16723805d1654))





## [0.4.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.4.0...@comparethemarketau/manor-modal@0.4.2) (2020-08-19)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.4.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.4.0...@comparethemarketau/manor-modal@0.4.1) (2020-08-19)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.4.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.3.1...@comparethemarketau/manor-modal@0.4.0) (2020-08-18)


### Features

* **all:** remove manorprovider from components, fix typography props ([#365](https://github.com/comparethemarketau/manor-react/issues/365)) ([9297cf7](https://github.com/comparethemarketau/manor-react/commit/9297cf72e8a7fe8762ec0dadf07d026aa88cbb44))





## [0.3.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.3.0...@comparethemarketau/manor-modal@0.3.1) (2020-08-18)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.3.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.2.4...@comparethemarketau/manor-modal@0.3.0) (2020-08-17)


* Fix/global styles micro ui (#359) ([bcf8ce9](https://github.com/comparethemarketau/manor-react/commit/bcf8ce92ba170a51113a4022728da22f47a6a768)), closes [#359](https://github.com/comparethemarketau/manor-react/issues/359)


### BREAKING CHANGES

* Global styles have been removed from ManorProvider, as such typography is not
styled by default. Import the Font component once in your application (or macro-ui) and use the
Typography component to render text, and the Link component to render links.

* fix(fonts): fix fonts dependencies

* fix(deps): fix provider dependencies

* fix(feedback): fixes based on PR feedback

Co-authored-by: PSJgit <psjj.dev@gmail.com>





## [0.2.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.2.3...@comparethemarketau/manor-modal@0.2.4) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.2.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.2.2...@comparethemarketau/manor-modal@0.2.3) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.2.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.2.1...@comparethemarketau/manor-modal@0.2.2) (2020-08-07)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.2.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.2.0...@comparethemarketau/manor-modal@0.2.1) (2020-07-24)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# [0.2.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.10...@comparethemarketau/manor-modal@0.2.0) (2020-07-20)


### Features

* **providers:** remove LayerEvent/Toast providers from manorprovider ([#348](https://github.com/comparethemarketau/manor-react/issues/348)) ([461d724](https://github.com/comparethemarketau/manor-react/commit/461d72498fca1aca9de0056a27d1a3d17a89ea77))


### BREAKING CHANGES

* **providers:** drawers and modals are no longer managed by manor, users will need to use the
zIndex prop (added to both) to do so particularly when multiple layered components are rendered
simultaneously. Additionally, you will need to implement your own keyboard bindings and focus
trapping to ensure conformance to the design system.





## [0.1.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.9...@comparethemarketau/manor-modal@0.1.10) (2020-07-14)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.8...@comparethemarketau/manor-modal@0.1.9) (2020-06-21)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.7...@comparethemarketau/manor-modal@0.1.8) (2020-06-17)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.6...@comparethemarketau/manor-modal@0.1.7) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.5...@comparethemarketau/manor-modal@0.1.6) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.4...@comparethemarketau/manor-modal@0.1.5) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.3...@comparethemarketau/manor-modal@0.1.4) (2020-06-15)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.2...@comparethemarketau/manor-modal@0.1.3) (2020-06-12)

**Note:** Version bump only for package @comparethemarketau/manor-modal





## [0.1.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.1...@comparethemarketau/manor-modal@0.1.2) (2020-06-11)


### Bug Fixes

* **toasts:** move manor-provider to peerDeps for notifications ([#332](https://github.com/comparethemarketau/manor-react/issues/332)) ([c15fee4](https://github.com/comparethemarketau/manor-react/commit/c15fee4368510c1d7be9a1b75856fcc2a990ab21))





## [0.1.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-modal@0.1.0...@comparethemarketau/manor-modal@0.1.1) (2020-06-11)

**Note:** Version bump only for package @comparethemarketau/manor-modal





# 0.1.0 (2020-06-11)


### Features

* **packages:** migrated to monorepo ([920a45e](https://github.com/comparethemarketau/manor-react/commit/920a45ec4b40a19de32f39f29693cbe1b1f314ae))
