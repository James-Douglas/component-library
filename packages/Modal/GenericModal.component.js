/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { useId } from '@comparethemarketau/manor-hooks';
import { Overlay } from '@comparethemarketau/manor-overlay';
import FocusTrap from 'focus-trap-react';
import {
  StyledAlignment, StyledModal,
} from './GenericModal.styles';

// eslint-disable-next-line react/prop-types
const Wrapper = ({ trapFocus, focusTrapOptions, children }) => {
  if (trapFocus) {
    return (
      <FocusTrap focusTrapOptions={focusTrapOptions}>
        {children}
      </FocusTrap>
    );
  }
  return <>{children}</>;
};

const GenericModal = ({
  id: propsId,
  visible,
  className,
  children,
  overlay,
  overlayOpacity,
  handleOverlayClick,
  zIndex,
  trapFocus,
  focusTrapOptions,
}) => {
  const id = useId(propsId);
  return (
    <>
      {overlay && <Overlay visible={visible} opacityLevel={overlayOpacity} handleClick={handleOverlayClick} zIndex={zIndex} />}
      {visible
      && (
        <Wrapper trapFocus={trapFocus} focusTrapOptions={focusTrapOptions}>
          <StyledAlignment visible={visible} zIndex={zIndex}>
            <StyledModal id={id} className={className}>
              {children}
            </StyledModal>
          </StyledAlignment>
        </Wrapper>
      )}
    </>
  );
};

GenericModal.propTypes = {
  /**
   * Unique identifier for the modal
   */
  id: PropTypes.string,
  /**
   * Bool for modal opening/closing. Handled via state in the parent component
   */
  visible: PropTypes.bool,
  /**
   * Extend styles with additional classNames
   */
  className: PropTypes.string,
  /**
   * Render an overlay over the remainder of the screen (blocking interaction)
   */
  overlay: PropTypes.bool,
  /**
   * Defines the opacity of the overlay, if rendered. (0 to 1)
   */
  overlayOpacity: PropTypes.number,
  /**
   * Called when the overlay is clicked
   */
  handleOverlayClick: PropTypes.func,
  /**
   * The child content of the Modal
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  /**
   * zIndex for the modal & overlay (if using)
   */
  zIndex: PropTypes.number,
  /**
   * Traps focus within the modal
   */
  trapFocus: PropTypes.bool,
  /**
   * Options to be passed to the focus-trap library
   * Common use case: By default the library deactivates the trap when the Escape key is pressed, for our
   * use case we likely want to disable that with focusTrapOptions: { escapeDeactivates: false }
   * https://github.com/focus-trap/focus-trap#focustrap--createfocustrapelement-createoptions
   */
  focusTrapOptions: PropTypes.object,
};

GenericModal.defaultProps = {
  id: null,
  visible: false,
  className: '',
  children: '',
  overlay: false,
  overlayOpacity: 0.7,
  handleOverlayClick: null,
  zIndex: 999999,
  trapFocus: false,
  focusTrapOptions: {},
};

export default GenericModal;
