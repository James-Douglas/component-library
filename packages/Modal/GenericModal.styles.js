import styled, { keyframes } from 'styled-components';

export const StyledAlignment = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  pointer-events: none;
  z-index: ${({ zIndex }) => zIndex};
`;

const animateIn = keyframes`
  from {
    transform: translateY(-20%);
    opacity: 0;
  }
  to {
    transform: translateY(0);
    opacity: 1;
  }
`;

export const StyledModal = styled.div`
  background: ${({ theme }) => theme.colors.white};
  box-shadow: ${({ theme }) => theme.boxShadow.lg};
  z-index: inherit;
  pointer-events: auto;
  position: relative;
  animation-name: ${animateIn};
  animation-duration: .3s;
`;
