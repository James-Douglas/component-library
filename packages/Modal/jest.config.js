const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Modal',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Modal',
  displayName: 'Modal',
  rootDir: '../..',
};
