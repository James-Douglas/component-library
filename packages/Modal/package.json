{
  "name": "@comparethemarketau/manor-modal",
  "version": "0.6.1",
  "description": "Manor Modal component",
  "main": "lib/index.cjs.js",
  "author": "Compare The Market",
  "license": "Proprietary",
  "scripts": {
    "build": "rollup -c '../../rollup.config.js'",
    "tdd": "jest --coverage --watchAll",
    "test": "cross-env JEST_SILENT_REPORTER_DOTS=true jest --reporters=jest-silent-reporter --coverage --watchAll",
    "test:ci": "cross-env CI=true jest",
    "prepare": "cp package.json lib/ && cp README.md lib/ && cp CHANGELOG.md lib/"
  },
  "publishConfig": {
    "registry": "https://nexus.comparethemarket.com.au/repository/platform-packages/",
    "access": "public",
    "tag": "latest"
  },
  "dependencies": {
    "@comparethemarketau/manor-button": "^0.5.12",
    "@comparethemarketau/manor-grid": "^0.6.9",
    "@comparethemarketau/manor-hooks": "^0.2.7",
    "@comparethemarketau/manor-overlay": "^0.3.0",
    "@comparethemarketau/manor-separator": "^0.4.6",
    "@comparethemarketau/manor-typography": "^0.7.0",
    "@comparethemarketau/manor-utils": "^0.2.4",
    "@fortawesome/fontawesome-svg-core": "^1.2.26",
    "@fortawesome/pro-regular-svg-icons": "^5.12.0",
    "@fortawesome/react-fontawesome": "^0.1.8",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@comparethemarketau/manor-input": "^0.9.1"
  },
  "peerDependencies": {
    "@comparethemarketau/manor-provider": "^0.5.9",
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "styled-components": "^5.0.0"
  }
}
