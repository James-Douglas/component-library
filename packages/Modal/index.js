import Modal from './Modal.component';
import WizardModal from './WizardModal.component';
import GenericModal from './GenericModal.component';

export { Modal, GenericModal, WizardModal };
