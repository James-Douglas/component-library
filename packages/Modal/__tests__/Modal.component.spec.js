import React from 'react';
import 'jest-styled-components';
import { Input } from '@comparethemarketau/manor-input';
import {
  render, fireEvent,
} from '../../../testUtils';
import Modal from '../Modal.component';

/* Modal
–––––––––––––––––––––––––––––––––––––––––––––––––– */

describe('Modal', () => {
  it('renders with minimal props', () => {
    const { baseElement } = render(<Modal id="test-modal" visible />);
    const modal = baseElement.querySelector('#test-modal');
    expect(modal).toBeInTheDocument();
    expect(modal).toMatchSnapshot();
  });

  it('renders with scroll lock', () => {
    const { baseElement, rerender } = render(
      <Modal id="test-modal" visible disableBodyScrollLock={false} trackingLabel="test">
        <div style={{ height: '100vh' }} />
      </Modal>,
    );

    expect(baseElement).toHaveStyle('overflow: hidden');

    rerender(
      <Modal id="test-modal" visible disableBodyScrollLock trackingLabel="test">
        <div style={{ height: '100vh' }} />
      </Modal>,
    );
    expect(baseElement).not.toHaveStyle('overflow: hidden');

    rerender(
      <Modal id="test-modal" visible={false} disableBodyScrollLock={false} trackingLabel="test">
        <div style={{ height: '100vh' }} />
      </Modal>,
    );
    expect(baseElement).not.toHaveStyle('overflow: hidden');
  });

  it('does not render if visible={false}', () => {
    const { container } = render(<Modal id="test-modal" visible={false} />);
    const modal = container.querySelector('#test-modal');
    expect(modal).not.toBeInTheDocument();
  });

  it('renders children and renders other components', () => {
    const { baseElement } = render(
      <Modal id="test-modal" visible>
        <div id="test-content">
          <p>some test text</p>
        </div>
        <Input id="test-child-input" handleChange={() => {}} />
      </Modal>,
    );

    const modal = baseElement.querySelector('#test-modal');
    const testDiv = modal.querySelector('#test-content');
    const input = modal.querySelector('#test-child-input');

    expect(input).toBeInTheDocument();
    expect(testDiv).toBeInTheDocument();
    expect(modal).toBeInTheDocument();
  });

  it('accepts and fires a close function as a prop', () => {
    const closeCb = jest.fn();
    const { baseElement } = render(<Modal id="test-modal" visible handleClose={closeCb} />);
    const modal = baseElement.querySelector('#test-modal');
    const closeBtn = modal.querySelector('.icon-close');

    fireEvent.click(closeBtn, { button: 0 });
    expect(closeCb).toHaveBeenCalled();
  });

  it('can be extended with additional classes', () => {
    const { baseElement } = render(<Modal id="test-modal" visible className="test test-1" />);
    const class1 = baseElement.querySelector('.test');
    const class2 = baseElement.querySelector('.test-1');

    expect(class1).toBeInTheDocument();
    expect(class2).toBeInTheDocument();
  });

  it('renders primary button but not secondary', () => {
    const primaryClick = jest.fn();
    const { baseElement } = render(<Modal id="test-modal" visible primaryActionTitle="Primary Button" handlePrimaryActionClick={primaryClick} />);
    const modal = baseElement.querySelector('#test-modal');
    const primaryButton = modal.querySelector('#primary-btn');

    expect(primaryButton).toBeInTheDocument();

    fireEvent.click(primaryButton, { button: 0 });
    expect(primaryClick).toHaveBeenCalled();
  });

  it('renders with supplementary bar', () => {
    const { baseElement } = render(<Modal id="test-modal" visible showSupplementaryBar supplementaryBarOptions={{ showRequestCall: true }} />);
    const modal = baseElement.querySelector('#test-modal');
    const supplementaryBar = modal.querySelector('#supplementary-bar');

    expect(supplementaryBar).toBeInTheDocument();
  });
});
