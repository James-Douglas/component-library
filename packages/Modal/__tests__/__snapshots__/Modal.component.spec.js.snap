// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders with minimal props 1`] = `
.c4 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 1.6rem;
  padding-left: 1.6rem;
  box-sizing: border-box;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 1.6rem;
  padding-left: 1.6rem;
  box-sizing: border-box;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0;
  max-width: 120rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -1.6rem;
  margin-left: -1.6rem;
  box-sizing: border-box;
  margin-bottom: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  background: #f6f6f6;
  box-shadow: 0 1.5rem 3rem 0 rgba(0,0,0,0.11),0 0.5rem 1.5rem 0 rgba(0,0,0,0.08);
  z-index: inherit;
  max-width: 768px;
  min-width: 350px;
  width: 768px;
  min-height: 30rem;
  height: 80%;
  pointer-events: auto;
  position: relative;
  -webkit-animation-name: bMDMvv;
  animation-name: bMDMvv;
  -webkit-animation-duration: .3s;
  animation-duration: .3s;
}

.c1 {
  background: #FFFFFF;
  padding: 1.9rem 2.4rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 h3 {
  margin: 0;
}

.c7 {
  cursor: pointer;
  z-index: inherit;
  color: #757575;
  font-size: 2.4rem;
}

.c8 {
  padding: 2rem 2.4rem 0;
  height: calc(100% - 6.8rem);
}

.c9 {
  height: calc(100% - 8rem);
  overflow: hidden scroll;
}

@media only screen and (min-width:0em) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:64em) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:75em) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:0em) {
  .c4 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:48em) {
  .c4 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:64em) {
  .c4 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:75em) {
  .c4 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:0em) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:48em) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:64em) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:75em) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:0em) {
  .c6 {
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c6 {
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    display: block;
  }
}

@media only screen and (min-width:64em) {
  .c6 {
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    display: block;
  }
}

@media only screen and (min-width:75em) {
  .c6 {
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    display: block;
  }
}

@media only screen and (min-width:0em) {
  .c6 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:48em) {
  .c6 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:64em) {
  .c6 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:75em) {
  .c6 {
    margin-left: 0%;
  }
}

@media only screen and (min-width:0em) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:48em) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:64em) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media only screen and (min-width:75em) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0 
    
  "
  id="test-modal"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3 row "
      >
        <div
          class="c4 
        
      "
        >
          <div
            class="c5"
          >
            
          </div>
        </div>
        <div
          class="c6 
        
      "
        >
          <div
            aria-label="Close Modal"
            aria-pressed="false"
            class="c7 icon-close"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times fa-w-10 "
              data-icon="times"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c8"
  >
    <div
      class="c9"
    />
  </div>
</div>
`;
